nupic/embodied/samplers/single_vecworker_sampler.py

```diff
@@ -0,0 +1,118 @@
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+# ------------------------------------------------------------------------------
+import copy
+
+from garage.experiment.deterministic import get_seed
+from garage.sampler import FragmentWorker, LocalSampler
+from garage.sampler.worker_factory import WorkerFactory
+
+SEED = get_seed()
+
+
+class SingleVecWorkSampler(LocalSampler):
+    """Sampler class which contains 1 vectorized worker which contains all the envs.
+
+    The sampler need to be created either from a worker factory or from
+    parameters which can construct a worker factory. See the __init__ method
+    of WorkerFactory for the detail of these parameters.
+
+    Args:
+        agents (Policy or List[Policy]): Agent(s) to use to sample episodes.
+            If a list is passed in, it must have length exactly
+            `worker_factory.n_workers`, and will be spread across the
+            workers.
+        envs (Environment or List[Environment]): Environment from which
+            episodes are sampled. If a list is passed in, it must have length
+            exactly `worker_factory.n_workers`, and will be spread across the
+            workers.
+        worker_factory (WorkerFactory): Pickleable factory for creating
+            workers. Should be transmitted to other processes / nodes where
+            work needs to be done, then workers should be constructed
+            there. Either this param or params after this are required to
+            construct a sampler.
+        max_episode_length(int): Params used to construct a worker factory.
+            The maximum length episodes which will be sampled.
+        is_tf_worker (bool): Whether it is workers for TFTrainer.
+        seed(int): The seed to use to initialize random number generators.
+        worker_class(type): Class of the workers. Instances should implement
+            the Worker interface.
+        worker_args (dict or None): Additional arguments that should be passed
+            to the worker.
+
+    """
+
+    def __init__(
+            self,
+            agents,
+            envs,
+            *,  # After this require passing by keyword.
+            worker_factory=None,
+            max_episode_length=None,
+            is_tf_worker=False,
+            seed=SEED,
```

Comment:
SEED is global, but not a constant. The all caps standard in python is used for constants.